### PR TITLE
Implement translation to the High Level WCS objects in GlobalCoords

### PR DESCRIPTION
**Pull request by Cadair**
This adds support for parsing the "dropped_world_dimensions" dictionary from `SlicedLowLevelWCS` and `ExtraCoords` into a High level object as done in `astropy.wcs.wcsapi.HighLevelWCSMixin.pixel_to_world`.

### CODE REVIEW

File: ndcube/global_coords/global_coords.py [hunk 1]
```diff
@@ -24,6 +27,65 @@ def __init__(self, ndcube=None):
...
+                kwargs[key][attr] = world[i]
+            else:
+                while attr > len(args[key]) - 1:
+                    args[key].append(None)
+                args[key][attr] = world[i]
+
+        for key in default_order(components):
+            key_ele = [i for i, components in enumerate(components) if components[0] == key]
+            physical_types = [dropped_dimensions["world_axis_physical_types"][i] for i in key_ele]
+            physical_types = physical_types[0] if len(physical_types) == 1 else tuple(physical_types)
```
Comment (DanRyanIrish) [inline ndcube/global_coords/global_coords.py:66]: Add a comment explaining why the length 1 check is done.

File: ndcube/global_coords/global_coords.py [hunk 2]
```diff
@@ -24,6 +27,65 @@ def __init__(self, ndcube=None):
...
+            names = tuple([dropped_dimensions["world_axis_names"][i] or
+                           dropped_dimensions["world_axis_physical_types"][i] for i in key_ele])
+            names = names[0] if len(names) == 1 else names
```
Comment (DanRyanIrish) [inline ndcube/global_coords/global_coords.py:69]: Same as above

File: ndcube/global_coords/global_coords.py [hunk 3]
```diff
@@ -24,6 +27,65 @@ def __init__(self, ndcube=None):
...
+            elif len(rest) == 1:
+                klass_gen = rest[0]
+            else:
+                raise ValueError("Tuples in world_axis_object_classes should have length 3 or 4")
+
+            high_level_object = klass_gen(*args[key], *ar, **kwargs[key], **kw)
+
+            # Special case SkyCoord to get a pretty name
+            if isinstance(high_level_object, SkyCoord):
+                high_level_object.name
```
Comment (DanRyanIrish) [inline ndcube/global_coords/global_coords.py:83]: Presumably this should be set to a variable here.

**Review by DanRyanIrish — COMMENTED** [review]
Looking goof so far.  A couple of quick comments added.

File: ndcube/global_coords/global_coords.py [hunk 4]
```diff
@@ -33,19 +100,20 @@ def _all_coords(self):
...
-                    wcs_dropped[name] = (physical_type, val)
-                return {**wcs_dropped, **self._internal_coords}
-
-        return self._internal_coords
+            dropped_world = copy.deepcopy(self._ndcube.wcs.low_level_wcs.dropped_world_dimensions)
+            if dropped_world:
+                wcs_dropped = self._convert_dropped_to_internal(dropped_world)
+                all_coords.update(wcs_dropped)
+
+        # TODO: Implement dropped_world_dimensions on ExtraCoords
```
Comment (DanRyanIrish) [inline ndcube/global_coords/global_coords.py:111]: If not already done, raise issue (or put comment on existing issue) and add to 2.0 project.

File: ndcube/tests/test_ndcube.py [hunk 5]
```diff
@@ -149,10 +149,10 @@ def test_slicing_removed_world_coords(ndcube_3d_ln_lt_l):
 
     sndc = ndc[0, 0, :]
     all_coords = sndc.global_coords._all_coords
-    assert u.allclose(all_coords[lat_key][1], -0.00555556 * u.deg)
-    assert u.allclose(all_coords[lon_key][1], 0.00277778 * u.deg)
-    assert all_coords[lat_key][0] == lat_key
-    assert all_coords[lon_key][0] == lon_key
+    assert isinstance(all_coords["helioprojective"][1], SkyCoord)
```
Comment (DanRyanIrish) [inline ndcube/tests/test_ndcube.py:152]: Probably best to set `celestial_key = "helioprojective"` and use that to make it easier to adapt to future changes to this name or if frame of test WCS is changed.

**Review by DanRyanIrish — APPROVED** [review]

File: ndcube/global_coords/global_coords.py (hunk 4 above)
Comment (Cadair) [reply ndcube/global_coords/global_coords.py:111]: It is the remainder of #349

**Pull request merged**